### PR TITLE
Add Milone eTape level sensor example

### PR DESCRIPTION
**Pull request by RBerliner**
Add the Milone eTape level sensor example to SensESP

### CODE REVIEW

File: examples/milone_level_sensor/milone_level_sensor.cpp [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+class ETapeInterpreter : public CurveInterpolator
+{
+
+public:
+  ETapeInterpreter(String config_path = "") : CurveInterpolator(NULL, config_path)
+  {
+
+    // Populate a lookup table tp translate the ohm values returned by
+    // our level sensor to inches on the sensor scale
+    clearSamples();
```
Comment (mairas) [inline examples/milone_level_sensor/milone_level_sensor.cpp:23]: Deprecated - use `clear_samples` instead.

File: examples/milone_level_sensor/milone_level_sensor.cpp [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    // addSample(CurveInterpolator::Sample(knownOhmValue, sensorHeight));
+    addSample(CurveInterpolator::Sample(388., 8.5));
```
Comment (mairas) [inline examples/milone_level_sensor/milone_level_sensor.cpp:25]: Deprecated, use `add_sample` instead.

File: examples/milone_level_sensor/milone_level_sensor.cpp [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+  // Create a sensor that is the source of our data, that will be read every 500 ms.
+  // It's a Milone depth sensor that's connected to the ESP's AnalogIn pin.
+  // The AnalogIn pin on ESP8266 is always A0, but ESP32 has many pins that can be
+  // used for AnalogIn, and they're expressed here as the XX in GPIOXX.
+
+  uint8_t pin = A0;
+  uint read_delay = 500;
+
+  auto *pAnalogInput = new AnalogInput(pin, read_delay, kAnalogInConfigPath);
```
Comment (mairas) [inline examples/milone_level_sensor/milone_level_sensor.cpp:99]: Code style: `analog_input`.

File: examples/milone_level_sensor/milone_level_sensor.cpp [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+  const float scale = 1.0;
+  const float Vin = 5.0;
+  const float R2 = 100;
+  const uint samples = 10;
+
+  // Wire up the output of the analog input to the VoltageDividerR1 transform.
+  // and then output the results to the SignalK server.
+
+  // Comment out the following 3 lines to suppress the output of the eTape sensor resistance.
+  pAnalogInput->connectTo(new AnalogVoltage(1.0, 1.0, 0.))
```
Comment (mairas) [inline examples/milone_level_sensor/milone_level_sensor.cpp:123]: Code style/deprecation: Use `connect_to` instead.

**Review by mairas — COMMENTED** [review]
From my point of view, this looks like a very nice example, and a neat implementation. I only have a few comments related to the code style.

@tedenda has been working on the example tree structure; I'm summoning him to verify that the PR is conforms to his changes (to me looks like it would).

**Comment by RBerliner** [comment]
Attempted to replace connectTo with connect_to and addSample with add_sample. This doesn't work.

Apparently the CurveInterpolator and AnalogInput classes have not yet been converted to GoogleStyle.

If I convert these classes to Google Style, it is going to break a lot of code.

**Comment by RBerliner** [comment]
maybe not. I need to delete .pio and try agaain

**Comment by RBerliner** [comment]
Fixed the naming issues that you identified.

File: examples/milone_level_sensor/README.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+fluid level height x is given by:
+
+##       x = (1385 - R)/140 - 1.  
+        
+The model calculation is shown in the table and in the graph. 
+
+## Software Implementation
+
+The supporting software for the Milone eTape is based on SensESP and is an extension of the AnalogInput example. Because of the way the sensor has been connected to the ESP8266 
+(measuring the voltage across the fixed reference resistor) it was necessary to add the VoltageDividerR1 class to SensESP. In addition, code has been added to the AnalogInput example to output intermediate results from the sensor level calculation. Rather than implement the conversion from eTape resistance to x as a linear transform, the CurveInterpolator class has been used. An additional record in the table: `addSample(CurveInterpolator::Sample(1600., 1.0));` has been added so that noise leading to resistances greater than 1386 Ohm gives a reasonable result.
```
Comment (tedenda) [inline examples/milone_level_sensor/README.md:58]: addSample --> add_sample

File: examples/milone_level_sensor/milone_level_sensor.cpp [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+  Serial.begin(115200);
+
+  // A small delay and one debugI() are required so that
+  // the serial output displays everything
+  delay(100);
+  Debug.setSerialEnabled(true);
+#endif
+  delay(100);
+  debugI("Serial debug enabled");
+
```
Comment (tedenda) [inline examples/milone_level_sensor/milone_level_sensor.cpp:63]: Debug code has been updated on all examples.
Please use this now:
```
#ifndef SERIAL_DEBUG_DISABLED
  SetupSerialDebug(115200);
#endif
```

**Review by tedenda — COMMENTED** [review]
Found two things that you can update.

File: examples/milone_level_sensor/README.md (hunk 5 above)
Comment (RBerliner) [reply examples/milone_level_sensor/README.md:58]: This has already been fixed

File: examples/milone_level_sensor/milone_level_sensor.cpp (hunk 6 above)
Comment (RBerliner) [reply examples/milone_level_sensor/milone_level_sensor.cpp:63]: Before I push these changes, is this what you want?

#ifndef SERIAL_DEBUG_DISABLED
  SetupSerialDebug(115200);
#endif
  delay(100);
  debugI("Serial debug enabled");

Comment (RBerliner) [reply examples/milone_level_sensor/milone_level_sensor.cpp:56]: nevermind, I fugured it out.

**Comment by RBerliner** [comment]
Fixed SERIAL_DEBUG_DISABLED

File: examples/milone_level_sensor/milone_level_sensor.cpp [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+class ETapeInterpreter : public CurveInterpolator
+{
+
+public:
+  ETapeInterpreter(String config_path = "") : CurveInterpolator(NULL, config_path)
+  {
+
+    // Populate a lookup table tp translate the ohm values returned by
+    // our level sensor to inches on the sensor scale
```
Comment (ba58smith) [inline examples/milone_level_sensor/milone_level_sensor.cpp:22]: "to translate", not "tp translate".

File: examples/milone_level_sensor/README.md (hunk 5 above)
Comment (mairas) [reply examples/milone_level_sensor/README.md:58]: Can you verify? Namely, the change in README.md line 58?

**Comment by RBerliner** [comment]
My typo production nearly equals my desired text production. I got the tp and the addSample->add_sample

**Comment by mairas** [comment]
I believe all raised concerns have been addressed. Thanks a lot for the very nice real-life example! Merging.

**Pull request merged**